### ISSUE TITLE
Recording tutorial data and checking the number of files produced

### ISSUE THREAD

**Issue by adarshp** [ml4ai/tomcat#98]
Recording tutorial data
-----------------------

We might as well record data from the tutorial mission in addition to the main mission. So we should update the `run_session` and `run_tutorial` script to include calls to `ffmpeg` to record the player audio, screen video, webcam video, and system audio when the tutorial mission is running.

Right now there is a function called `start_ffmpeg_recording` in  `tools/run_session` - that can be used as a starting point for implementing this. 

The crudest way to implement this would be to create a copy of the function with different filenames for the relevant output and log files , maybe something called `start_ffmpeg_recording_tutorial` or similar, and call it just before `tools/run_tutorial`. 

A more elegant way would be to abstract out the common elements from both the functions, creating a single function that sets the output and log file names based on whether the tutorial mission is being recorded or the main mission.

See https://www.gnu.org/software/bash/manual/html_node/Shell-Functions.html for a reference on Bash functions.

Let's have the directory structure for the session be something like this (for example)

```
tomcat/data/participant_data/
└── session_2020_04_15_15_40_46
    ├── main
    │   ├── discrete_events.json
    │   ├── malmo_data.txt
    │   ├── player_audio.wav
    │   ├── screen_video.mpg
    │   ├── system_audio.wav
    │   └── webcam_video.mpg
    └── tutorial
        ├── discrete_events.json
        ├── malmo_data.txt
        ├── player_audio.wav
        ├── screen_video.mpg
        ├── system_audio.wav
        └── webcam_video.mpg
```

I haven't included the self-reports in this list above since (i) they are not always generated, and (ii) the way they are implement right now, they can have varying file names.

Checking the number of files produced
----------------------------------------

It would be good to have the `run_session` script double-check that all the files that were supposed to be created in the session have, in fact, been created. This involves putting a few lines of code at the end of the `run_session` script that checks whether the directory structure and the files have been correctly generated, and if not, produce appropriate error messages.

**Comment by adarshp** [comment]
@dl932 I was working on refactoring some of the scripts in the `robustify_scripts` branch yesterday, and it made sense to implement the recording of the tutorial data in that effort (see commit https://github.com/ml4ai/tomcat/commit/a4abe6aa20c5339bf81c751bcf280b129fdcd768). So no need to do that part. But you can still go ahead and implement the checking of the number of files produced. You may want to create a new branch off of the `robustify_scripts` branch.

**Comment by dl932** [comment]
@adarshp I implemented the checking codes based on the robustify_scripts branch. Two folders would be produced and they are named by "mission_"$CURRENT_MISSION"". The tutorial one is named by "mission_0". The main mission is getting its name from the parameter passing "do_mission ${MAIN_MISSION}". But how I can I know the value of "${MAIN_MISSION}"? Should I just use the variable to be the fold name? Then I used this branch and failed to run. Does it mean I will test until the bugs are fixed in this branch?

**Comment by adarshp** [comment]
@dl932 `MAIN_MISSION` has a default value of 1 - you can see this on line 51 of `tools/configuration_helpers`.

I'm not sure why the branch is not working for you - I had tested it just yesterday and it worked. Make sure you are up-to-date with the latest version of the branch, and if you are, then try to find out where the bug is (if there is one).

**Comment by dl932** [comment]
> @dl932 `MAIN_MISSION` has a default value of 1 - you can see this on line 51 of `tools/configuration_helpers`.
> 
> I'm not sure why the branch is not working for you - I had tested it just yesterday and it worked. Make sure you are up-to-date with the latest version of the branch, and if you are, then try to find out where the bug is (if there is one).

Got it. Thanks!

**Comment by dl932** [comment]
@adarshp The first error message on my mac is [ xdpyinfo:  unable to open display ""] and it is from function "__configure_screen_capture()" in file "configuration_helpers". It seems that its running "xdpyinfo" on my mac and thus this message prompted. I manually ran the xdpyinfo executable and got the same message.
![bug1](https://user-images.githubusercontent.com/58101904/79910939-64764780-83d4-11ea-8b2a-9a34b5093151.jpg)
<img width="536" alt="image" src="https://user-images.githubusercontent.com/58101904/79911362-0dbd3d80-83d5-11ea-9ac4-e835e89e5f2b.png">

**Comment by adarshp** [comment]
@dl932 From the last answer here: https://stackoverflow.com/questions/37826094/xt-error-cant-open-display-if-using-default-display?answertab=active#tab-top it seems like you might need to reboot your computer after installing `xdpyinfo` - can you try that?

**Comment by dl932** [comment]
Should I install xdpyinfo? But I tried rebooting. I got the same error message.

**Comment by dl932** [comment]
@adarshp Should I install xdpyinfo? But I tried rebooting. I got the same error message.

**Comment by adarshp** [comment]
Closed by #132
